src/main/java/com/google/devtools/build/lib/skyframe/SkyframeActionExecutor.java

```diff
@@ -1537,8 +1538,15 @@ private boolean checkOutputs(
           if (output.isTreeArtifact()) {
             reportOutputTreeArtifactErrors(action, output, reporter, e);
           } else {
-            // Are all exceptions caught due to missing files?
-            reportMissingOutputFile(action, output, reporter, output.getPath().isSymbolicLink(), e);
+            if (e instanceof NotASymlinkException) {
+              reporter.handle(Event.error(
+                  action.getOwner().getLocation(),
+                  String.format("declared output '%s' is not a symlink", output.prettyPrint())));
```

Comment:
How about "Output '%s' was declared as a symlink, but is a '%s'"?

Reply:
Good point, will update the corresponding PR if possible.